### DISCUSSION TITLE
Localize .NET MAUI application

### DISCUSSION THREAD

**Discussion by giscus** [VladislavAntonyuk/vladislavantonyuk.github.io#71] [Announcements]
# Localize .NET MAUI application

This article describes how to localize .NET MAUI application and switch the language without restarting the app.

https://vladislavantonyuk.github.io/articles/Localize-.NET-MAUI-application/

**Comment by cgomezm** [comment]
Hi Vlad, thanks for the tutoria. 

I've been following this up and everything works good in debug mode but, when running in release im not getting the text value but the binding itself "Microsoft.Maui.Controls.Binding". Do you know what could i be missing here? 

This is my Translate extension
[ContentProperty(nameof(Key))]
    internal class TranslateExtension : IMarkupExtension
    {
        public string Key { get; set; }

        public object ProvideValue(IServiceProvider serviceProvider)
        {
            var binding = new Binding
            {
                Mode = BindingMode.OneWay,
                Path = $"[{Key}]",
                Source = Translator.Instance
            };

            return binding;
        }
    }   

Thanks

**Reply by cgomezm** [reply]
I know it can be fixed if i return the acutal string value like this
public object ProvideValue(IServiceProvider serviceProvider)
        {
            return Translator.Instance[Key];
        }
but wonder why there is the behavior difference between debug and release

**Reply by VladislavAntonyuk** [reply]
Hello, this a .NET MAUI issue https://github.com/dotnet/maui/issues/18697

**Reply by cgomezm** [reply]
Thanks Vlad!
